lib/dialects/mysql/query/compiler.js

```diff
@@ -18,6 +18,47 @@ class QueryCompiler_MySQL extends QueryCompiler {
     this._emptyInsertValue = '() values ()';
   }
 
+  // Compiles an `insert` query, allowing for multiple
+  // inserts using a single query statement.
+  insert() {
+    let sql = QueryCompiler.prototype.insert.call(this);
+    if (sql === '') return sql;
+
+    const { ignore, merge, insert } = this.single;
+    if (ignore) sql = sql.replace('insert into', 'insert ignore into');
+    if (merge) sql += this._merge(merge.updates, insert);
+
+    return sql;
+  }
+
+  _merge(updates, insert) {
+    let sql = ' on duplicate key update ';
+    if (updates) {
+      const updateData = this._prepUpdate(updates);
+      if (typeof updateData === 'string') {
+        sql += updateData;
+      } else {
+        sql += updateData.join(',');
+      }
+
+      return sql;
+    } else {
+      const insertData = this._prepInsert(insert);
+      if (typeof insertData === 'string') {
+        throw new Error(
+          'If using merge with a raw insert query, then updates must be provided'
+        );
+      }
+
+      sql += insertData.columns
+        .map((column) => this.formatter.wrapString(column))
```

Comment:
wrapIdentifier?

Reply:
This method works just like wrapString, but without checking for including ' as '

Reply:
Ah yeah, I guess inserts would never contain `as`.